### PR TITLE
Improvement to plugin system docs

### PR DESCRIPTION
**Pull request by giovannipizzi**
Fix some things that were not particularly clear in the documentation for the new plugin system.

### CODE REVIEW

File: docs/source/developer_guide/plugins/overview.rst [hunk 1]
```diff
@@ -59,19 +55,19 @@ It is important to keep in mind that, even though there is very little that is i
...
-A plugin may / should not
+A plugin may/should not:
 
-* change the database schema AiiDA uses
-* monkey patch anything within the ``aiida`` namespace nor the namespace itself
-* use functionality of aiida or it's classes, which is marked as implementation detail, examples include functions, methods and classes starting with an underscore, '`_`'.
-* circumvent data proveniency in any way
+* change the database schema AiiDA uses;
+* monkey patch anything within the ``aiida`` namespace or the namespace itself;
+* in general, use functionality of AiiDA or its classes which is marked as implementation detail: examples include functions, methods and classes starting with an underscore symbol: `_`.
```
Comment (fernandogargiulo1986) [inline docs/source/developer_guide/plugins/overview.rst]: either:
use a functionality of AiiDA ... which is marked
or:
use funcionalities of AiiDA ... which are marked

File: docs/source/developer_guide/plugins/overview.rst [hunk 2]
```diff
@@ -59,19 +55,19 @@ It is important to keep in mind that, even though there is very little that is i
 
 Failure to comply with the points below will probably mean that your plugin will not be able to be listed as an officially endorsed plugin.
 
-A plugin may / should not
+A plugin may/should not:
```
Comment (fernandogargiulo1986) [inline docs/source/developer_guide/plugins/overview.rst]: I think the meaning you have in mind is best conveyed by:
A plugin cannot (should not):
Ex:
Users may not use the querybuilder if they know how to use SQL.
Users cannot push changes direcctly into aiida_core/develop

File: docs/source/developer_guide/plugins/update_plugin.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
-Updating an Existing Plugin
-===========================
+Updating an Existing Plugin (or creating a new one)
+===================================================
 
-This section will take you through the steps needed to create a python package containing your plugin, which can then be easily developped, version controlled distributed and updated separately from aiida.
+This section will walk you through the steps needed to create a python package containing your plugin, which can then be easily developed, version controlled distributed and updated separately from AiiDA.
```
Comment (fernandogargiulo1986) [inline docs/source/developer_guide/plugins/update_plugin.rst]: revise the noun word "version controlled distributed". Maybe you want to say: "distributed through version controlled"

File: docs/source/developer_guide/plugins/contributing.rst [hunk 4]
```diff
@@ -52,11 +59,13 @@ Interfaces
 Pluginloader (aiida/common/pluginloader.py)
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The pluginloader relies on reentry (PYPI package) to find and load entry points. reentry has been added to setup_requires for aiida in order to enable scanning for existing plugins when aiida is installed. If for some reason reentry should be uninstalled, the plugin system will fall back on pkg_resources from setuptools, which is slower.
+The pluginloader relies on `reentry`_ (PYPI package) to find and load entry points. ``reentry`` has been added to setup_requires for AiiDA in order to enable scanning for existing plugins when AiiDA is installed. If for some reason reentry is uninstalled or is not found, the plugin system will fall back on ``pkg_resources`` from setuptools, which is slower.
```
Comment (giovannipizzi) [inline docs/source/developer_guide/plugins/contributing.rst]: In the link it's not possible http://docutils.sourceforge.net/FAQ.html#is-nested-inline-markup-possible but I'll fix the rest

**Pull request merged**